File: backend/app/agents/search_agent.py
from langchain.agents import create_agent
from app.core.llm import get_llm
from app.tools.web_tools import web_search

def build_search_agent():
    llm = get_llm()

    return create_agent(
        model=llm,
        tools=[web_search],
        system_prompt="You are a research assistant. Find the most relevant and recent information using web search."
    )

File: syncai/src/lib/agents/searchAgent.ts
import { getModel } from "../llm";
import { tavilySearch } from "../tools/webTools";

export async function runSearchAgent(topic: string): Promise<string> {
  const rawResults = await tavilySearch(topic);

  const model = getModel();
  const result = await model.generateContent(
    `You are a research assistant. Synthesize these web search results about "${topic}" into a concise, organized summary. Preserve key facts and any important URLs.

Search Results:
${rawResults}

Provide a clear synthesis:`,
  );

  return result.response.text();
}
